rllib/core/rl_trainer/rl_trainer.py

```diff
@@ -178,6 +177,48 @@ def compute_loss(
         # should find a way to allow them to specify single-agent losses as well,
         # without having to think about one extra layer of hierarchy for module ids.
 
+        loss_total = None
+        results_all_modules = {}
+        for module_id in fwd_out:
+            module_batch = batch[module_id]
+            module_fwd_out = fwd_out[module_id]
+
+            module_results = self._compute_loss_per_module(
+                module_id, module_batch, module_fwd_out
+            )
+            results_all_modules[module_id] = module_results
+            loss = module_results[self.TOTAL_LOSS_KEY]
```

Comment:
just a bit confusing, why would module_results ever contain TOTAL_LOSS_KEY at this point?

Reply:
that's the convention. User needs to return their loss under that key.

Reply:
they can optionally include other metrics they want to log.